### PR TITLE
Integration start

### PR DESCRIPTION
**Pull request by zachgersh**
introduces the acceptance tests that I mentioned wanting to spec out.

Benefits:
- All in go!
- Can of course be run anywhere
- Mock server can be replaced with a real one if / when we like
- Only one external dependency `spec`
- Random test ordering and they run in parallel
- Can redirect the test output if we need to via the test configuration

If you want to read more about [spec](https://github.com/sclevine/spec) and the "whys" behind it.

I have a bit to finish with the auth command and then I am thinking we could discuss this a bit after that.

### CODE REVIEW

**Comment by zachgersh** [comment]
also, yes I know I have not included this in the `make test` command as of yet - intentionally!

File: acceptance/account_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+const accountOutput string = `
+Email                     Droplet Limit    Email Verified    UUID                                        Status
+sammy@digitalocean.com    25               true              b6fr89dbf6d9156cace5f3c78dc9851d957381ef    active
+`
+
+const ratelimitOutput string = `
+Limit    Remaining    Reset
+200      199          2019-08-09 14:24:41 -0700 PDT
```
[the remark this answers is not in the document]
Comment (zachgersh) [reply acceptance/account_test.go]: darn, yeah good catch on the timezone :/

**Comment by zachgersh** [comment]
> A few small comment/suggestions.
> 
> Overall I wonder how this approach will end up working for acceptance tests that involve the user's local config, a remote Kubernetes config, or any other dependencies beyond the API.
> 
> As we discussed on Slack, I wish it was easier to run a single test case (ie, be able to run individual tests in a way that's supported by editors that have a `go test` integration), but I understand that's a tradeoff of the `spec` library.

Local config wise, we can do a save / replace of the original since we know the test will be destructive OR we can retrofit the code a bit to allow for users to specify another location for the configs.

Focusing of tests is how you run a singular test in the `spec` world. So, you'd have to do something like a `it.Focus` from your editor instead. I hope that would work be a bit faster than copying the test name?

**Comment by zachgersh** [comment]
@hilary - I suggest we don't do this only because it will slightly complicate test running as you will have to pick whether you are running a) unit tests b) integration tests c) both.

We can mix them together via `build tags` so that you can do the decisions I  mentioned above but I would rather just leave it in a separate directory.

**Comment by hilary** [comment]
> @hilary - I suggest we don't do this only because it will slightly complicate test running as you will have to pick whether you are running a) unit tests b) integration tests c) both.
> 

@zachgersh How does leaving them in a separate directory help? 

In any case, as a developer, I want `make unit`, `make integration` and `make test`. Build tags seems like a small price to pay to localize (for example) the account related logic.

**Comment by bentranter** [comment]
@zachgersh @hilary If you want to run them alongside the code and don't want to complicate running unit tests vs. integration, you can put the integration tests behind the `testing.Short` boolean and decide whether to run them or not with the `-short` flag, as I've mentioned before.

**Comment by zachgersh** [comment]
@hilary - I think a separate directory shows that they aren't actually demonstrates that they are at the highest level of the testing pyramid, dealing with a built main artifact. The tests that are mixed in, which I love by the way, are located directly next to their concern and that makes sense for a unit test since you will more than likely have both your unit and implementation open at the same time.

There are plenty of projects that share this sort of ethos (integration, e2e, and anything besides unit live in a specific other directory):

1. https://github.com/openshift/origin (has a test directory)
2 https://github.com/kubernetes/kubernetes/tree/master/test (same sort of deal with a test directory)

@bentranter - thanks for mention `short` as well, I neglected to mention that.

**Comment by Verolop** [comment]
Belated comment: 
I think all the recent projects that I've been involved in use @zachgersh proposed structure, and it's popular for big distributed systems (which I'm aware is not our case rn) for different reasons, particularly:
1. Organisation: the people (or team) writing the unit tests are not the same people (team) in charge of the testing framework (let testing be e2e, integration, etc.), so they're allowed to be treated as separate code entities (and responsibilities).
2. Flexibility: for purposes like testing in production or non-functional purposes, it allows enough flexibility to treat those files as separate pieces of code, without the "overhead" of having to select what type of tests you want to work with every time. 
It's worth noting tho, that for those use cases unit tests are close to useless, since they don't guarantee the behaviour of the interconnections, so it makes sense to have another approach :)

**Pull request merged**